src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,222 @@
+        last.next = node;
+        last = node;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
 
+        if (size == index) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (index == 0) {
+            node = new Node<>(null, value, first);
+            first.prev = node;
+            first = node;
+            size++;
+            return;
+        }
+        node = first;
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
+        Node<T> addNode = new Node<>(node.prev, value, node);
+        node.prev.next = addNode;
+        node.prev = addNode;
+        size++;
+
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        if (index == size - 1) {
+            return last.item;
+        }
+        iterationFromTheFirstElement(index);
+        iterationFromTheLastElement(index);
+        return node.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> returnNode = null;
+        if (index == 0) {
+            node = new Node<>(null, value, first.next);
+            returnNode = first;
+            first.next.prev = node;
+            first = node;
+            return returnNode.item;
+        }
+        if (size - index > size / 2) {
+            returnNode = first;
+            for (int i = 1; i <= index; i++) {
+                returnNode = returnNode.next;
+            }
+        } else {
+            node = last;
+            for (int i = size - 1; i >= index; i--) {
+                returnNode = returnNode.prev;
+            }
+        }
+        node = new Node<>(returnNode.prev, value, returnNode.next);
+        returnNode.prev.next = node;
+        returnNode.next.prev = node;
+        return returnNode.item;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> removedNode = null;
+        if (index == 0) {
+            node = first;
+            if (size == 1) {
+                first = last = null;
+                size--;
+                return node.item;
+            }
+            first.next.prev = null;
+            first = first.next;
+            size--;
+            return node.item;
+        }
+        if (index == size - 1) {
+            node = last;
+            last = last.prev;
+            last.next = null;
+            size--;
+            return node.item;
+        }
+        iterationFromTheFirstElement(index);
+        iterationFromTheLastElement(index);
+        node.prev.next = node.next;
+        node.next.prev = node.prev;
+        size--;
+        return node.item;
```

Comment:
same and I'am pretty sure that you can use this in other remove method

Reply:
done